src/worker_operation.py

```diff
@@ -0,0 +1,40 @@
+from time import time
+from traceback import format_exc
+
+
+class WorkerOperation:
```

Comment:
FYI - "worker operation" is not a common term.
also, I would imagine adding some fail-track callbacks to the constructor so in case of failure you can run these callbacks (like checking which phase failed, logging, monitoring etc..)